exercises/exercise4/checkSpecialCases.xquery

```diff
@@ -0,0 +1,149 @@
+        return $result
+    )
+};
+
+(: Check, if last pit is from the current player :)
+declare function local:getIsPitOfCurrentPlayer($lastPit as xs:string, $currentPlayer as xs:string) {
+    if ($currentPlayer = "player1") then
+        if (fn:starts-with($lastPit, "house1")) then (
+            let $result := true
+            return $result
+        )
+        else (
+            let $result := false
+            return $result
+        )
+    else if ($currentPlayer = "player2") then (
+        if (fn:starts-with($lastPit, "house2")) then (
+            let $result := true
+            return $result
+        )
+        else (
+            let $result := false
+            return $result
+        )
+    )
+    else (
+        (: Do nothing :)
+    )
+};
+
+(: Check, if last pit was empty before :)
+declare function local:getWasLastPitEmpty($lastPit as xs:string) {
+    if ($lastPit/numOfSeeds/text() = "1") then (
+        let $result := true
+        return $result
+    )
+    else (
+        let $result := false
+        return $result
+    )
+};
+
+(: Get opposite pit house number :)
+declare function local:getOppositePit($lastPit as xs:string) {
+    if (fn:starts-with($lastPit, "house1")) then (
+        let $houseNumber1x := (fn:substring-after($lastPit, "house")) + 10
+        return fn:concat("house", $houseNumber1x)
+    )
+    else if (fn:starts-with($lastPit, "house2")) then (
+        let $houseNumber2x := (fn:substring-after($lastPit, "house")) - 10
+        return fn:concat("house", $houseNumber2x)
+    )
+    else (
+        (: Do nothing :)
+    )
+};
+
+(: Check, if opposite pit contains any seeds :)
+declare function local:getOppositePitHasSeeds($oppositePit as xs:string) {
+    if ($oppositePit/numOfSeeds/text() > 0) then (
+        let $result := true
+        return $result
+    )
+    else (
+        let $result := false
+        return $result
+    )
+};
+
+
+(: ---- Setter functions ---- :)
+(: Take last seed and opposite seeds, then move them to own seed store :)
+declare updating function local:fillPlayersStore($relevantStore as xs:string, $lastPit as xs:string) {
+    (: Get current store seed count :)
+    (: Take seeds from last pit, add them into the relevant store :)
+    (: Take seeds from opposite pit, add them into the relevant store :)
+    if ($relevantStore = "store1" or "store2") then (
+        let $currentStoreSeedCount := local:getStoreSeedCount($currentPlayer)
+        let $lastPitValue := $gameInstance/$lastPit/numOfSeeds/text()
+        let $oppositePitValue := $gameInstance/$oppositePit/numOfSeeds/text()
+        let $newStoreValue := $currentStoreSeedCount + $lastPitValue + $oppositePitValue
+        return (replace value of node $gameInstance/$relevantStore/numOfSeeds/text() with $newStoreValue)
+    )
+    else (
+        (: Do nothing :)
+    )
+};
+
+declare updating function local:setPitsToZero($lastPit as xs:string) {
+    let $pathToLastPit := $gameInstance/$lastPit/numOfSeeds/text()
```

Comment:
This will not work, use: let $pathToLastPit := $gameInstance/*[name()=$lastPit]/numOfSeeds/text()
Has to be modified throughout the code.